src/stores/store.ts

```diff
@@ -0,0 +1,23 @@
+import { create } from 'zustand';
+import { persist, createJSONStorage } from 'zustand/middleware';
+
+type Store = {
+  currentView: string;
+  setCurrentView: (view: string) => void;
+};
+
+const useStore = create<Store>()(
+  persist(
+    (set) => ({
+      currentView: 'landing',
+      setCurrentView: (view: string) => set({ currentView: view }),
+    }),
```

Comment:
what is going on with persist here? that looks interesting

Reply:
Persists lets us persist a store's state across page reloads (if using session storage) and application restarts (if using local storage).